shared/widgets/CharacterCreation/index.tsx

```diff
@@ -85,36 +85,45 @@ class CharacterCreation extends React.Component<CharacterCreationProps, any> {
   }
 
   create = () => {
-    // try to create...
-    let model: CharacterCreationModel = {
-      name: (this.refs['name-input'] as any).value,
-      race: this.props.racesState.selected.id,
-      gender: this.props.gender,
-      faction: this.props.factionsState.selected.id,
-      archetype: this.props.playerClassesState.selected.id,
-      shardID: this.props.shard,
-      attributes: this.props.attributesState.attributes.reduce((acc: any, cur: AttributeInfo) => {
-        if (cur.type !== attributeType.PRIMARY) return acc;
-        if (typeof acc.name !== 'undefined') {
-          let name = acc.name;
-          let val = acc.allocatedPoints;
-          acc = {};
-          acc[name] = val;
-        }
-        if (typeof acc[cur.name] === 'undefined' || isNaN(acc[cur.name])) {
-          acc[cur.name] = cur.allocatedPoints;
-        } else {
-          acc[cur.name] += cur.allocatedPoints;
-        }
-        return acc;
-      }),
+    // validate name
+    const modelName = (this.refs['name-input'] as any).value.trim();
+    let errors: any = [];
+    if (modelName.length < 2 || modelName.length > 20) errors.push('A character name must be between 2 and 20 characters in length.');
```

Comment:
The length of 2 is after normalization. So, the name ( O' ) would pass your validation, but not the server.  ( O'B ) would pass as it has a min of 2 after normalization.

Additionally, I think I neglected to mention this in the trello card, you can not have consecutive special characters.  ( O``B ) is invalid, ( O--B ) is invalid  ( O-B' ) is valid.

Lastly, the first character must be a letter.  ( -OB ) is invalid  ( O-B ) is valid